pkg/codegen/internal/test/testdata/resource-property-overlap/go-extras/tests/go_test.go

```diff
@@ -0,0 +1,77 @@
+package tests
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"resource-property-overlap/example"
+)
+
+func TestArrayOutputIndex(t *testing.T) {
```

Comment:
This actually times out. I can't figure it out why. @pgavlin 

Before your PR it was failing with a different error.

From what I can tell `example.RecArray{rec}.ToRecArrayOutput().Index(pulumi.Int(0)).` may be coming back with a *Rec that is not initialized as a resource, so it's URN is never populated.

Accidental struct copy?